### ISSUE TITLE
Multi-Config CMake Presents do not work with VS Code using CMake extension

### ISSUE THREAD

**Issue by prince-chrismc** [conan-io/conan#13908]
Conan version: 2.0.4
Platform, OS: MacOS
CMake version: 3.26.0
CMake VS Code extension and its version: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools 1.15.8

***

Only the default "configure" preset is loaded and detected how ever it's not possible to build and test with the debug or release presets. Everything is done with "Default" which in my case was release that I ran first

You can see only one option is preset
![image](https://github.com/conan-io/conan/assets/16867443/d9375e1b-5cd1-481e-a6e7-bc96fead5ca2)

I was not able to find anything suggest that picking a build or test preset when opening the project

```
7932  conan new cmake_lib --define name=hello -d version=0.1
7941  conan install . -c tools.cmake.cmaketoolchain:generator="Ninja Multi-Config"
7942  conan install . -c tools.cmake.cmaketoolchain:generator="Ninja Multi-Config" -s build_type=Debug
7943  cmake --preset conan-default
7944  cmake --build --preset conan-release
7945  cmake --build --preset conan-debug
```

**Comment by memsharded** [comment]
Not sure what could be failing there.
I am in Windows. My VS Code load the ``conan-default`` configure preset correctly, and I can also switch and build the ``conan-release`` and ``conan-debug`` build presets.

Please report:
- Conan version
- Platform, OS
- CMake version
- CMake VS Code extension and its version

**Comment by prince-chrismc** [comment]
After reading the documentation, I found you need to go into the command pallet to enable it, its not there by default on Mac perhaps. It was not in the UI until after that and it's only in the toolbar 

https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md

Looking over the code there are places where it does name matching https://github.com/microsoft/vscode-cmake-tools/blob/8fc40e8e5d2a00290958e5de6942184bbb94a4fd/src/presetsController.ts#L723 🤔 
https://github.com/microsoft/vscode-cmake-tools/blob/8fc40e8e5d2a00290958e5de6942184bbb94a4fd/src/presetsController.ts#L684

**Comment by memsharded** [comment]
So it is working now? Could we close the issue then?

**Comment by prince-chrismc** [comment]
I think it deserves some documentation, how the multi config presets are setup by Conan and how they should be used is missing.

If you didn't know there was three of them then you would not be able to solve this.

**Comment by memsharded** [comment]
Multi-config presets are explained and used in this docs: https://docs.conan.io/2/examples/tools/cmake/cmake_toolchain/build_project_cmake_presets.html. Anything specific missing in that doc page?

**Comment by prince-chrismc** [comment]
Yes! It gives the CLI which is great but I feel it needs a small reference section for the different type of presets that are created.

What's missing is the significance of the different presets - there's no mention they are being used ("notice here it's the other one" would help)

This way there's a self help option to connect the dots to cmake and other tooling.

- what presets are created and when
- How and when each preset should be used
- what it's like forgetting to pass the preset (I think this where the novice is mostly to make a mistake so it would help if it is possible to document)

****

(I am making this up I have not 100% figured it out what conan is doing and when)

"In single configuration cmaketoolchain creates only one preset that is used for configuration. You need to specify this when configuring cmake, however there are no build or testing preset so you do not" 

"For multi config generator, there are 3 preset each of which needs to be passed at the correct step to CMake
- Conan-default is created first and is the configuration preset. Using this preset for building and testing will always correspond to the first configuration you `conan installed`
- conan release is the build and test preset that needs to be called when doing `cmake --build` or `cmake --test`
- conan-debug preset...

You need to make sure to specify the correct preset at each stage of your workflow. When no preset is given the behavior is `conan-default`
